### PR TITLE
Issue #3229816 by chmez: As a CM+ I want to be able to see biggest trends in content

### PR DESCRIPTION
**Pull request by ghost**
## Problem
As a CM+ I want to be able to see biggest trends in content.

## Solution
- for custom content list block add new sorting option:
  - Most comments and likes (summed up likes and comments)
- for custom content list block add new filtering option:
  - by date: x amount of days
  - should be based on comments or likes created by given amount of days
  - should be shown only when trending is selected as sorting option

## Issue tracker
- https://www.drupal.org/project/social/issues/3229816
- https://getopensocial.atlassian.net/browse/OEC-20

## How to test
- [x] Enable the "Social Content Block" module and log in as administrator.
- [x] When you navigate to the "/block/add/custom_content_list" page then you should see the "Comments and likes" option in the "Sorting" field.
- [x] When you select the "Most comments and likes" option in the "Sorting" field then you should see the "Amount of days" field.
- [x] You should see the "90" value in the "Amount of days" field by default.

## Release notes
Add a new sorting option to the custom content list block by the sum of likes and comments.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Review by ribel — APPROVED** [review]
I have tested changes on the default Custom content list block and on also Dashboard as CM and all works as expected. Code changes look good as well.

File: modules/social_features/social_content_block/src/ContentBlockBase.php [hunk 1]
```diff
@@ -19,11 +19,21 @@ abstract class ContentBlockBase extends PluginBase implements ContentBlockPlugin
...
+        'limit' => FALSE,
+      ],
       'most_commented' => 'Most commented',
       'most_liked' => 'Most liked',
-      'last_interacted' => 'Last interacted',
+      'last_interacted' => [
+        'label' => 'Last interacted',
+        'limit' => FALSE,
+      ],
+      'trending' => 'Comments and likes',
```
Comment (ribel) [inline modules/social_features/social_content_block/src/ContentBlockBase.php:36]: @OH17L As we discussed with Jessica could you rename it to the `Most comments and likes` for now?

Comment (OH17L) [reply modules/social_features/social_content_block/src/ContentBlockBase.php]: Done.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2474-qvovtbgawrl04f2vlxalbigdeke0hdtj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61264e39378c5f70b98934ac

**Review by ribel — APPROVED** [review]

**Comment by ribel** [comment]
🍒 picked to 10.3.x

**Pull request merged**